modules/installation-azure-service-principal.adoc

```diff
@@ -0,0 +1,124 @@
+
+[id="installation-azure-service-principal_{context}"]
+= Creating a service principal
+
+Because {product-title} and its installation program must create Microsoft Azure
+resources through Azure Resource Manager, you must create a service principal
+to represent it.
+
+.Prerequisites
+
+* Install or update the link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-yum?view=azure-cli-latest[Azure CLI].
+* Install the `jq` package.
+//* Your Azure account has the `User Access Administrator` role.
+
+.Procedure
+
+. Log into the Azure CLI:
++
+----
+$ az login
+----
++
+Log into Azure in the web console by using your credentials.
+
+. If your Azure account uses subscriptions, ensure that you are using the right one.
+.. To view your account details, run the following command:
++
+----
+$ az account show
+{
+  "environmentName": "AzureCloud",
+  "id": "9bab1460-96d5-40b3-a78e-17b15e978a80",
+  "isDefault": true,
+  "name": "Subscription Name", <1>
+  "state": "Enabled",
+  "tenantId": "6057c7e9-b3ae-489d-a54e-de3f6bf6a8ee",
+  "user": {
+    "name": "you@example.com",
+    "type": "user"
+  }
+}
+----
+<1> Ensure that the value of the name parameter is the name of your subscription.
+
+.. You can change the subscription to use by obtaining the subscription ID
+from the Azure portal and setting it in the CLI:
++
+----
+$ az account set -s <subscription_ID>
+----
+
+.. If you changed the subscription ID, display your account information again:
++
+----
+$ az account show
+
+{
+  "environmentName": "AzureCloud",
+  "id": "33212d16-bdf6-45cb-b038-f6565b61edda",
+  "isDefault": true,
+  "name": "Subscription Name",
+  "state": "Enabled",
+  "tenantId": "8049c7e9-c3de-762d-a54e-dc3f6be6a7ee",
+  "user": {
+    "name": "you@example.com",
+    "type": "user"
+  }
+}
+----
+
+. Record the values of the `tenantId` and `id` parameters from the previous
+output. You need these values during {product-title} installation.
+
+. Create the service principal for your account:
++
+----
+$ az ad sp create-for-rbac --role "User Access Administrator" -role Contributor --name <service_principal> <1>
+Changing "<service_principal>" to a valid URI of "http://<service_principal>", which is the required format used for service principal names
+Retrying role assignment creation: 1/36
+Retrying role assignment creation: 2/36
+Retrying role assignment creation: 3/36
+Retrying role assignment creation: 4/36
+{
+  "appId": "8bd0d04d-0ac2-43a8-928d-705c598c6956",
+  "displayName": "<service_principal>",
+  "name": "http://<service_principal>",
+  "password": "ac461d78-bf4b-4387-ad16-7e32e328aec6",
+  "tenant": "6048c7e9-b2ad-488d-a54e-dc3f6be6a7ee"
+}
+
+----
+<1> Specify the name to assign to the service principal.
+
+. Record the values of the `appId` and `password` parameters from the previous
+output. You need these values during {product-title} installation.
+
+. Grant additional permissions to the service principal. The service principal
```

Comment:
For this step, if you don't have admin access you have to request an admin approve this. But if you do, does it just get applied automatically?

Reply:
I am not sure - I only have access to an account that does not have the correct permissions. @ingvagabund, do you know someone who has admin access and could test this out?

Reply:
I don't think it will get approved automatically in any case. @coverprice @cuppett might now?